integrated_channels/moodle/exporters/content_metadata.py

```diff
@@ -114,3 +114,10 @@ def transform_end(self, content_metadata_item):
         if end_date:
             return int(parse(end_date).replace(tzinfo=timezone.utc).timestamp())
         return None
+
+    def _apply_delete_transformation(self, content_metadata_item):
+        """
+        Specific transformations required for "deleting/hiding" a course on a Moodle.
+        """
+        content_metadata_item['visible'] = 0  #hide a course on moodle - instead of true delete
```

Comment:
this is close, but instead of a dictionary value I think we want the full string that moodle requires (ie 'courses[0][visible]'  https://github.com/openedx/edx-enterprise/pull/1881/commits/48df722785f920a63d45397da1053d69376190f7#diff-4b9552fde027242408a12442ddd2aba02cc0f856d3e42575f03df72524169667R415).

Reply:
We can also add `'wsfunction': 'core_course_update_courses'` 🤔

Reply:
re- https://github.com/openedx/edx-enterprise/pull/1881/commits/7182b831646f3ff281e1131343b052259c33a4d4#diff-4b9552fde027242408a12442ddd2aba02cc0f856d3e42575f03df72524169667R406

Reply:
1. Yeah I also added the same(`courses[0][visible]`) at first attempt but turned out it appeared as `courses[0]courses[0][visible]`_(notice the double courses[0])_ in `delete_content_metadata` payload `serialized_data`.

Reply:
2. Now, If I also add `wsfunction: 'core_course_update_courses'` - it will automatically become `courses[0][wsfunction]` in `delete_content_metadata` payload `serialized_data`; which isn't the format that moodle supports I guess 🤔 `wsfunction` shouldn't be part of `courses[]`

Reply:
Here's serialized_data log in delete_content_metadata. 

See even though I didn't added courses[0] but it already had courses[0]['visible']. Maybe ContentMetadataExporter::export is doing that.

![image](https://github.com/openedx/edx-enterprise/assets/83753341/f40641d9-c88d-4d97-8f59-6541fb6dd9fd)